rest/route/project.go

```diff
@@ -218,6 +219,10 @@ func (h *projectIDPatchHandler) Factory() gimlet.RouteHandler {
 // Parse fetches the project's identifier from the http request.
 func (h *projectIDPatchHandler) Parse(ctx context.Context, r *http.Request) error {
 	h.projectID = gimlet.GetVars(r)["project_id"]
+	user := MustHaveUser(ctx)
+	if user != nil {
```

Comment:
You don't have to check this--there is supposed to be a user, and in the case that the user is nil `MustHaveUser` panics.

Reply:
(and it's ok to panic in the context of a request because it just returns a 500 and it's totally fine.) 

also it should be impossible to get here without having users set.

Reply:
I added in this case because online tests were failing if I didn't. 
https://evergreen.mongodb.com/test_log/5de6ca9cc9ec444bb581919f#L593

Reply:
Change the test, not the code--you probably just need to add a user to the handler.

Reply:
I fixed this here: https://github.com/evergreen-ci/evergreen/pull/2926/commits/16d90a4e2457d4ffae6ce924594bdb3c6c143b56